iOS/Projects/Features/Onboarding/Sources/Presentaion/ViewModel/OnboardingViewModel.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  OnboardingViewModel.swift
+//  OnboardingFeature
+//
+//  Created by MaraMincho on 11/29/23.
+//  Copyright © 2023 kr.codesquad.boostcamp8. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+// MARK: - OnboardingViewModelInput
+
+public struct OnboardingViewModelInput {
+  let shouldPresentMapAuthorizationPublisher: AnyPublisher<Void, Never>
+  let shouldPresentHealthAuthorizationPublisher: AnyPublisher<Void, Never>
+}
+
+public typealias OnboardingViewModelOutput = AnyPublisher<OnboardingState, Never>
+
+// MARK: - OnboardingState
+
+public enum OnboardingState {
+  case idle
+  case shouldPresentMapAuthorization(OnboardingScenePropertyDTO)
+  case shouldPresentHealthAuthorization(OnboardingScenePropertyDTO)
+  case finish
+  case errorState(Error)
+}
+
+// MARK: - OnboardingViewModelRepresentable
+
+public protocol OnboardingViewModelRepresentable {
+  func transform(input: OnboardingViewModelInput) -> OnboardingViewModelOutput
+}
+
+// MARK: - OnboardingViewModel
+
+public final class OnboardingViewModel {
+  // MARK: - Properties
+
+  private var useCase: OnboardingImageLoadUseCaseRepresentable
+  public init(useCase: OnboardingImageLoadUseCaseRepresentable) {
+    self.useCase = useCase
+  }
+
+  private var subscriptions: Set<AnyCancellable> = []
+}
+
+// MARK: OnboardingViewModelRepresentable
+
+extension OnboardingViewModel: OnboardingViewModelRepresentable {
+  public func transform(input: OnboardingViewModelInput) -> OnboardingViewModelOutput {
+    subscriptions.removeAll()
+
+    let presentMapAuth: OnboardingViewModelOutput = input.shouldPresentMapAuthorizationPublisher
+      .tryMap { [weak self] _ -> OnboardingState in
+        guard let self else {
+          return .errorState(OnboardingViewModelError.didNotInitViewModel)
+        }
+        guard let dto = useCase.mapOnboardingImage() else {
+          return .errorState(OnboardingViewModelError.nilValue)
+        }
+        return .shouldPresentMapAuthorization(dto)
+      }
+      .catch { error in return Just(OnboardingState.errorState(error)) }
+      .eraseToAnyPublisher()
+
+    let presentHealthAuth: OnboardingViewModelOutput = input.shouldPresentHealthAuthorizationPublisher
+      .tryMap { [weak self] _ -> OnboardingState in
+        guard let self else {
+          return .errorState(OnboardingViewModelError.didNotInitViewModel)
+        }
+        guard let dto = useCase.healthOnboardingImage() else {
+          return .errorState(OnboardingViewModelError.nilValue)
+        }
+        return .shouldPresentHealthAuthorization(dto)
+      }
+      .catch { error in return Just(OnboardingState.errorState(error)) }
+      .eraseToAnyPublisher()
```

Comment:
p2: tryMap에서 throw를 하지 않고 return을 해주고 있습니다. tryMap을 올바르게 사용하려면 throw를 사용하셔야 합니다!
그리고 weak self 없이 흐름을 만들 수 있을 것 같아요.

```suggestion
    let presentMapAuth: OnboardingViewModelOutput = input.shouldPresentMapAuthorizationPublisher
      .map(useCase.mapOnboardingImage)
      .tryMap { dto in
        guard let dto else { throw OnboardingViewModelError.nilValue }
        return .shouldPresentMapAuthorization(dto)
      }
      .catch { error in return Just(.errorState(error)) }
      .eraseToAnyPublisher()

    let presentHealthAuth: OnboardingViewModelOutput = input.shouldPresentHealthAuthorizationPublisher
      .map(useCase.healthOnboardingImage)
      .tryMap { dto in
        guard let dto else { throw OnboardingViewModelError.nilValue }
        return .shouldPresentHealthAuthorization(dto)
      }
      .catch { error in return Just(.errorState(error)) }
      .eraseToAnyPublisher()
```

Reply:
헉 여기서 왜 tryMap을 썻죠..? 부끄럽네요 ㅎㅎ ☺️🙈